packages/core/src/Cardano/types/ProtocolParameters.ts

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable no-use-before-define */
+import { Slot } from '../';
+
+export interface ProtocolVersion {
+  major: number;
+  minor: number;
+  patch?: number;
+}
+export type CostModels = CostModel[];
+
+export type CostModel = number[];
+
+export interface Prices {
+  memory: number;
+  steps: number;
+}
+export interface ExUnits {
+  memory: number;
+  steps: number;
+}
+
+export interface ValidityInterval {
+  invalidBefore?: Slot;
+  invalidHereafter?: Slot;
+}
+
+export interface ProtocolParametersBabbage {
```

Comment:
Don't think all of those parameters have to be nullable. Since this has 'Babbage' in it's name, it should probably have all parameters as non-nullable. Alternatively, we could have a `ProtocolParameters` type that has original parameters as non-nullable and parameters added in later eras as nullable.

Changes in this PR didn't change this (was already like that), so this could be addressed as a separate task.

Reply:
I agree we should be reworking this. Ogmios types the properties as nullable since it's in context of a protocol update, which can include one or many alterations to the current blockchain params. We should have a base that's non-nullable then a version to model the update type.

Reply:
ADP-2165